circuits/voteverifier/vote_verifier.go

```diff
@@ -147,16 +219,11 @@ func (c *VerifyVoteCircuit) checkCircomProof(api frontend.API) error {
 	// check that the circom witness only contains a single public input
 	// (the hash of all the public-private inputs)
 	api.AssertIsEqual(len(c.CircomPublicInputsHash.Public), 1)
-	// hash the circom public-private inputs and compare them with the unique
-	// public input of the circom circuit
-	circomInputs := c.circomInputs(api)
-	h, err := mimc7.NewMiMC(api)
-	if err != nil {
+	// verify the hash of the circom public-private inputs
+	if err := checkInnerInputHash(api, c.CircomPublicInputsHash.Public[0],
+		c.circomInputs(api)...); err != nil {
 		return err
```

Comment:
We should wrap these errors, or when they appear in the logs we won't be able to know where it came from.